### DISCUSSION TITLE
In an expo managed app, how do I set "View controller-based status bar appearance -> YES"?

### DISCUSSION THREAD

**Discussion by chriscoomber** [react-navigation/react-navigation#11902] [Q&A] [answered]
I'm trying to use the `statusBarStyle: 'light'` option on one of my screens. The [documentation](https://reactnavigation.org/docs/native-stack-navigator/#statusbarstyle) for this says:

> Requires setting View controller-based status bar appearance -> YES (or removing the config) in your Info.plist file.

Without setting this, I understandably see the error:

> If you want to change the appearance of status bar, you have to change UIViewControllerBasedStatusBarAppearance key in the Info.plist to YES

However, I'm writing a managed **expo** app, so I can't edit the Info.plist file directly. How do I make this change?

### Extra information

- [Simple repro](github.com/chriscoomber/react-navigation-status-bar-repro)

- I've seen that you can add Info.plist keys to `app.json`:
    ```json
    {
      "expo": {
        "ios": {
          "infoPlist": {
            "UIViewControllerBasedStatusBarAppearance": true
          }
        },
      }
    }
    ```
    - When is this actually applied? It doesn't seem to have any effect on **expo go**.
    - Do I need to switch to using **expo dev client**?
    
- I don't have expo-status-bar installed

**Comment by satya164** [comment, marked as answer]
Yes, you need to use expo dev client for such options to work.
